Task7.3

deseff opened the pull request:
Cześć Maciek,

wydaje mi się, że większość kodu napisałem poprawnie, ale pokazuje mi błąd w klasie Forum przy: ```
public List<ForumUser> getUserList() {
            return new ArrayList<>(theForumUsers);
        }
```
Podpowiesz, co mam zmienić?

deseff committed:
Module 7.3. - examples from the theory

deseff committed:
Module 7.3. - task

Zadanie: funkcyjny spacer po liście użytkowników forum
Twoim zadaniem jest stworzenie kolekcji składającej się z obiektów opisujących użytkowników forum. Następnie, przy pomocy Stream powinieneś przekształcić tę kolekcję w mapę zawierającą tych użytkowników forum, którzy mają nie mniej niż 20 lat, są mężczyznami i mają co najmniej jeden opublikowany post.

Aby zrealizować zadanie wykonaj:

Stwórz w katalogu src/main/java pakiet com.kodilla.stream.forumuser, a w nim klasę ForumUser
Klasa ForumUser powinna mieć następujące pola (użyj odpowiednich nazw angielskich):
unikalny identyfikator użytkownika (typu int)
nazwa użytkownika (typu String)
płeć (typu char), z wartościami ‘M’/’F’
data urodzenia (typu LocalDate)
ilość opublikowanych postów (typu int)
W tym samym pakiecie napisz klasę Forum, która będzie zawierała listę użytkowników, czyli listę obiektów ForumUser.
Klasa Forum powinna udostępniać metodę getUserList(), która będzie zwracała kolekcję z użytkownika forum
W metodzie main(String[] args) klasy StreamMain napisz program, który:
zainicjuje strumień Stream przy pomocy metody stream() kolekcji
odfiltruje tylko tych użytkowników, którzy są mężczyznami
odfiltruje tylko tych użytkowników, którzy mają co najmniej 20 lat
odfiltruje tylko tych użytkowników, którzy mają co najmniej jeden opublikowany post
przy pomocy kolektora utworzy mapę par, w której rolę klucza będzie pełnił unikalny identyfikator użytkownika
wyświetli otrzymaną mapę wynikową

MARudnicki commented on kodilla-stream/src/main/java/com/kodilla/stream/forumuser/Forum.java:
```diff
@@ -0,0 +1,29 @@
+        }
+        public List<ForumUser> getUserList() {
+            return new ArrayList<>(theForumUsers);
+        }
```
nawiasy sie nie zgadzaja, stad kod sie nie kompiluje. Obecnie getUserList jest w scope konstruktora Forum

deseff committed:
Corrections applied
Module 7.3. - task

Zadanie: funkcyjny spacer po liście użytkowników forum
Twoim zadaniem jest stworzenie kolekcji składającej się z obiektów opisujących użytkowników forum. Następnie, przy pomocy Stream powinieneś przekształcić tę kolekcję w mapę zawierającą tych użytkowników forum, którzy mają nie mniej niż 20 lat, są mężczyznami i mają co najmniej jeden opublikowany post.

Aby zrealizować zadanie wykonaj:

Stwórz w katalogu src/main/java pakiet com.kodilla.stream.forumuser, a w nim klasę ForumUser
Klasa ForumUser powinna mieć następujące pola (użyj odpowiednich nazw angielskich):
unikalny identyfikator użytkownika (typu int)
nazwa użytkownika (typu String)
płeć (typu char), z wartościami ‘M’/’F’
data urodzenia (typu LocalDate)
ilość opublikowanych postów (typu int)
W tym samym pakiecie napisz klasę Forum, która będzie zawierała listę użytkowników, czyli listę obiektów ForumUser.
Klasa Forum powinna udostępniać metodę getUserList(), która będzie zwracała kolekcję z użytkownika forum
W metodzie main(String[] args) klasy StreamMain napisz program, który:
zainicjuje strumień Stream przy pomocy metody stream() kolekcji
odfiltruje tylko tych użytkowników, którzy są mężczyznami
odfiltruje tylko tych użytkowników, którzy mają co najmniej 20 lat
odfiltruje tylko tych użytkowników, którzy mają co najmniej jeden opublikowany post
przy pomocy kolektora utworzy mapę par, w której rolę klucza będzie pełnił unikalny identyfikator użytkownika
wyświetli otrzymaną mapę wynikową

MARudnicki reviewed (changes requested):
W module kodilla-stream brakuje pliku build.gradle

deseff committed:
build.gradle added
Corrections applied
Module 7.3. - task

Zadanie: funkcyjny spacer po liście użytkowników forum
Twoim zadaniem jest stworzenie kolekcji składającej się z obiektów opisujących użytkowników forum. Następnie, przy pomocy Stream powinieneś przekształcić tę kolekcję w mapę zawierającą tych użytkowników forum, którzy mają nie mniej niż 20 lat, są mężczyznami i mają co najmniej jeden opublikowany post.

Aby zrealizować zadanie wykonaj:

Stwórz w katalogu src/main/java pakiet com.kodilla.stream.forumuser, a w nim klasę ForumUser
Klasa ForumUser powinna mieć następujące pola (użyj odpowiednich nazw angielskich):
unikalny identyfikator użytkownika (typu int)
nazwa użytkownika (typu String)
płeć (typu char), z wartościami ‘M’/’F’
data urodzenia (typu LocalDate)
ilość opublikowanych postów (typu int)
W tym samym pakiecie napisz klasę Forum, która będzie zawierała listę użytkowników, czyli listę obiektów ForumUser.
Klasa Forum powinna udostępniać metodę getUserList(), która będzie zwracała kolekcję z użytkownika forum
W metodzie main(String[] args) klasy StreamMain napisz program, który:
zainicjuje strumień Stream przy pomocy metody stream() kolekcji
odfiltruje tylko tych użytkowników, którzy są mężczyznami
odfiltruje tylko tych użytkowników, którzy mają co najmniej 20 lat
odfiltruje tylko tych użytkowników, którzy mają co najmniej jeden opublikowany post
przy pomocy kolektora utworzy mapę par, w której rolę klucza będzie pełnił unikalny identyfikator użytkownika
wyświetli otrzymaną mapę wynikową
